src/apps/applets/service/applet_service.py

```diff
@@ -17,3 +23,17 @@ def get_prev_version(self, version: str):
         if int_version < int(self.INITIAL_VERSION.replace(".", "")):
             return self.INITIAL_VERSION
         return ".".join(list(str(int_version - self.VERSION_DIFFERENCE)))
+
+    async def exist_by_id(self, applet_id: int) -> bool:
+        return await AppletsCRUD().exist_by_id(applet_id)
+
+    async def delete_applet_by_id(self, user_id, applet_id: int):
+        await self._validate_delete_applet(user_id, applet_id)
+        await AppletsCRUD().delete_by_id(applet_id)
+
+    async def _validate_delete_applet(self, user_id, applet_id):
+        role = await UserAppletAccessService(user_id, applet_id).is_admin()
+        if not role:
+            raise DoesNotHaveAccess(
+                message="You do not have access to delete applet."
+            )
```

Comment:
No need to create an additional role variable

```suggestion
    async def _validate_delete_applet(self, user_id, applet_id):
        if not (await UserAppletAccessService(user_id, applet_id).is_admin()):
            raise DoesNotHaveAccess(
                message="You do not have access to delete applet."
            )
```